src/Data/DataSource.php

```diff
@@ -458,6 +459,29 @@ public static function resolve_user_role( $name ) {
 
 	}
 
+	/**
+	 * Resolve the avatar for a user
+	 *
+	 * @param int   $user_id ID of the user to get the avatar data for
+	 * @param array $args    The args to pass to the get_avatar_data function
+	 *
+	 * @return array|null|Avatar
+	 * @throws \Exception
+	 */
+	public static function resolve_avatar( $user_id, $args ) {
+
+		$avatar = get_avatar_data( absint( $user_id ), $args );
```

Comment:
It would be ideal if we could defer this and do one batch request, but upon initial investigation, it appears that there is no Batch API for Gravatar, so here we are, stuck with n+1 requests to Gravatar for the image 😭 😭